Create release notes for version-5.0.md (#2419)
TykTechnologies/tyk-docs: tyk-docs/content/release-notes/version-5.0.md

---
title: Tyk v4.3
menu:
  main:
    parent: "Release Notes"
weight: 1
---

# Major features

## Improved OpenAPI support

We have added some great features to the Tyk OAS API definition bringing it closer to parity with our Tyk Classic API and to make it easier to get on board with Tyk using your Open API workflows.

Tyk’s OSS users can now make use of extensive [**custom middleware**](https://tyk.io/docs/plugins/) options with your OAS APIs, to transform API requests and responses, exposing your upstream services in the way that suits your users and internal API governance rules. We’ve enhanced the Request Validation for Tyk OAS APIs to include parameter validation (path, query, headers, cookie) as well as the body validation that was introduced in Tyk 4.1.

> Tyk Dashboard has been enhanced with an all **new custom middleware UI** for Tyk OAS APIs, so **for the first time** you can configure and manage all of your Tyk API middleware from the Dashboard; this covers the full suite of custom middleware from pre- to post- and response plugins. We’ve got support for middleware bundles, Go plugins and Tyk Virtual Endpoints, all within the new improved Tyk Dashboard.

Reviewer: ```suggestion
Tyk Dashboard has been enhanced with **all the custom middleware options** for Tyk OAS APIs, so **for the first time** you can configure your custom middleware from the Dashboard; this covers the full suite of custom middleware from pre- to post- and response plugins. We’ve got support for middleware bundles, Go plugins and Tyk Virtual Endpoints, all within the new and improved Tyk Dashboard UI.
```

- via command line using tyk-sync
- 
To make it easier for our users to find their way to Universal Data Graph, we’ve also given it its own space in the Dashboard. From now on you can find UDG under Data Graphs section of the menu.

It also got a lot easier to turn a Kafka topic into a GraphQL subscription. Using our new Dashboard API endpoint, users will be able to transform their AsyncAPI documentation into Universal Data Graph definition with a single click. Support for OAS coming soon as well!

With this release we are also giving our users improved headers for GQL APIs. It is now possible to use context variables in request headers and persist headers needed for introspection separately for improved security.

Additionally we’ve added Dashboard support for introspection control on policy and key level. It is now possible to allow or block certain consumers from being able to introspect any graph while creating a policy or key via Dashboard.

# Changelog

## Tyk Gateway

### Deprecated
- LetsEncrypt

### Added

### Changed

### Fixed

## Tyk Dashboard

### Added

### Changed

### Fixed

# Updated Versions
Tyk Gateway 5.0
Tyk Dashboard 5.0

# Upgrade process

> Follow the [standard upgrade guide]({{< ref "/content/upgrading-tyk.md" >}}), there are no breaking changes in this release.

Reviewer: ```suggestion
Follow the [standard upgrade guide]({{< ref "upgrading-tyk.md" >}}), there are no breaking changes in this release.
```

> If you want switch from MongoDB to SQL, you can [use our migration tool]({{< ref "planning-for-production/database-settings/postgresql.md#migrating-from-an-existing-mongodb-instance" >}}), but keep in mind that it does not yet support the migration of your analytics data.

Reviewer: ```suggestion
If you want to switch from MongoDB to SQL, you can [use our migration tool]({{< ref "planning-for-production/database-settings/postgresql.md#migrating-from-an-existing-mongodb-instance" >}}), but keep in mind that it does not yet support the migration of your analytics data.
```

> If you want switch from MongoDB to SQL, you can [use our migration tool]({{< ref "/content/planning-for-production/database-settings/postgresql.md#migrating-from-an-existing-mongodb-instance" >}}), but keep in mind that it does not yet support the migration of your analytics data.

Reviewer: ```suggestion
If you want switch from MongoDB to SQL, you can [use our migration tool]({{< ref "planning-for-production/database-settings/postgresql.md#migrating-from-an-existing-mongodb-instance" >}}), but keep in mind that it does not yet support the migration of your analytics data.
```

{{< note success >}}
**Note**  

> Note: Upgrading the Golang version implies that all the Golang custom plugins that you are using need to be recompiled before migrating to 4.3 version of the Gateway. Check our docs for more details [Golang Plugins]({{< ref "/content/plugins/supported-languages/golang.md" >}}).

Reviewer: ```suggestion
Note: Upgrading the Golang version implies that all the Golang custom plugins that you are using need to be recompiled before migrating to v5.0 of the Gateway. Check our docs for more details [Golang Plugins]({{< ref "plugins/supported-languages/golang#upgrading-tyk" >}}).
```

### Added
- Support for OpenAPI request validation (including query params, headers and the rest of OAS rules)
- Transform request/response middleware for OpenAPI apis
- Custom middleware for OpenAPI apis
- Added a new API endpoint to manage versions for OpenAPI apis
- Improved Mock API plugin for OpenAPI
- Universal Data Graph and GraphQL APIs now support using context variables in request headers, allowing passing information it to your subgraphs
- Now you can control access to introspection on policy and key level

### Changed

### Fixed
- Fixed potential race when using distributed rate limiter

## Tyk Dashboard

### Added
- Numerous UX improvements
- New UI for custom middleware for OpenAPI apis
- Significantly improved OpenAPI versioning user experience
- It now possible to use PATCH method to modify OpenAPI apis via Dashboard API
- Now you can turn Kafka topic into GraphQL subscription by simply importing your AsyncAPI definition
- Way to control access to introspection on policy and key level

### Changed
- Universal Data Graph moved to a separate dashboard section

# Updated Versions

> Tyk Gateway 5.0

Reviewer: Can we add links to the images in docker hub?
